pkg/server/commands/listusers/list_users_rpc_test.go

```diff
@@ -1695,6 +1695,243 @@ func TestListUsersEdgePruning(t *testing.T) {
+		      define required_1: [user]
+		      define required_2: [user]
+		      define can_view: required_1 and required_2`,
+
+			tuples: []*openfgav1.TupleKey{
+				tuple.NewTupleKey("document:1", "required_1", "user:*"),
+				tuple.NewTupleKey("document:1", "required_2", "user:*"),
+
+				tuple.NewTupleKey("document:1", "required_1", "user:maria"),
+				tuple.NewTupleKey("document:1", "required_2", "user:maria"),
+			},
+			expectedUsers: []string{"user:maria"},
+		},
+
+		{
+			name: "with_multiple_wildcards_5",
+			req: &openfgav1.ListUsersRequest{
+				Object:   &openfgav1.Object{Type: "document", Id: "1"},
+				Relation: "can_view",
+				UserFilters: []*openfgav1.ListUsersFilter{
+					{
+						Type: "user",
+					},
+				},
+			},
+			model: `model
+		    schema 1.1
+		  type user
+		  type document
+		    relations
+		      define required_1: [user]
+		      define required_2: [user:*]
+		      define can_view: required_1 and required_2`,
+
+			tuples: []*openfgav1.TupleKey{
+				tuple.NewTupleKey("document:1", "required_1", "user:maria"),
+				tuple.NewTupleKey("document:1", "required_2", "user:*"),
+			},
+			expectedUsers: []string{"user:maria"},
+		},
+		{
+			name: "with_multiple_wildcards_6",
+			req: &openfgav1.ListUsersRequest{
+				Object:   &openfgav1.Object{Type: "document", Id: "1"},
+				Relation: "can_view",
+				UserFilters: []*openfgav1.ListUsersFilter{
+					{
+						Type: "user",
+					},
+				},
+			},
+			model: `model
+		    schema 1.1
+		  type user
+		  type document
+		    relations
+		      define required_1: [user]
+		      define required_2: [user]
+		      define can_view: required_1 and required_2`,
+
+			tuples: []*openfgav1.TupleKey{
+				tuple.NewTupleKey("document:1", "required_1", "user:maria"),
+				tuple.NewTupleKey("document:1", "required_1", "user:jon"),
+
+				tuple.NewTupleKey("document:1", "required_1", "user:will"),
+				tuple.NewTupleKey("document:1", "required_2", "user:will"),
+
+				tuple.NewTupleKey("document:1", "required_1", "user:*"),
+				tuple.NewTupleKey("document:1", "required_2", "user:*"),
+			},
+			expectedUsers: []string{"user:will"},
+		},
+		{
+			name: "with_multiple_wildcards_7",
+			req: &openfgav1.ListUsersRequest{
+				Object:   &openfgav1.Object{Type: "document", Id: "1"},
+				Relation: "can_view",
+				UserFilters: []*openfgav1.ListUsersFilter{
+					{
+						Type: "user",
+					},
+				},
+			},
+			model: `model
+		    schema 1.1
+		  type user
+		  type document
+		    relations
+		      define required_1: [user,user:*]
+		      define required_2: [user,user:*]
+		      define can_view: required_1 and required_2`,
+
+			tuples: []*openfgav1.TupleKey{
+				tuple.NewTupleKey("document:1", "required_1", "user:maria"),
+				tuple.NewTupleKey("document:1", "required_1", "user:jon"),
+				tuple.NewTupleKey("document:1", "required_1", "user:*"),
+				tuple.NewTupleKey("document:1", "required_2", "user:will"),
+			},
+			expectedUsers: []string{"user:will"},
+		},
```

Comment:
Could we add another test that walks some userset expansion stages with wildcards?

```
model
  schema 1.1

type user

type group
  relations
    define member: [user:*, user]

type document
  relations
    define group: [group]
    define viewer: [group#member] and member from group
    define can_view: viewer
```
- document:1#viewer@group:y#member
- document:1#group@group:x
- group:x#member@user:*
- group:y#member@user:jon

`ListUsers(document:1#can_view, user_filters=[{"type": "user"}])` should return ["user:jon"]